src/lazy/retry/index.tsx

```diff
@@ -0,0 +1,60 @@
+export type RetryPolicy = {
+  delay: number;
+  maxAttempts: number | undefined;
+};
+
+type AttemptContext = {
+  attemptsRemaining: number;
+};
+
+export type Retryable<T> = () => Promise<T>;
+
+const calculateDelay = (context: AttemptContext, policy: RetryPolicy) => {
+  const { delay, maxAttempts } = policy;
+  // Do not use a delay on the first retry, or when the configured delay is 0
+  if (context.attemptsRemaining === maxAttempts || delay === 0) {
+    return 0;
+  }
+
+  return delay;
```

Comment:
Should we multiply this by the number of attempts? Otherwise going to keep trying every 200ms with will spam network and console tabs quite aggressively (if big number of attemps)

Reply:
Implemented as a linear backoff, using the expected times, as discussed